sdk/storage/Azure.Storage.Blobs/src/BlockBlobWriteStream.cs

```diff
@@ -63,9 +72,9 @@ protected override async Task FlushInternal(bool async, CancellationToken cancel
 
             Response<BlobContentInfo> response = await _blockBlobClient.CommitBlockListInternal(
                 base64BlockIds: _blockIds,
-                blobHttpHeaders: default,
-                metadata: default,
-                tags: default,
+                blobHttpHeaders: _blobHttpHeaders,
+                metadata: _metadata,
+                tags: _tags,
```

Comment:
Do we need to set these on every `Flush` call?  Or just the first?  (And if it's only the very first, could we do it on the initial blob creation?)  I don't think it's a huge deal either way, but it might lead to slightly smaller payloads if folks are passing this stream to a library that's constantly flushing.

Reply:
That's a good idea. I overlooked empty blob creation part.

Reply:
I did an experiment. Last flush wins.